homeassistant/components/homematicip_cloud/__init__.py

```diff
@@ -211,6 +222,32 @@
         schema=SCHEMA_DEACTIVATE_VACATION,
     )
 
+    async def _set_active_climate_profile(service):
+        """Service to set the active climate profile."""
+        entity_id_list = service.data.get(ATTR_ENTITY_ID)
+        climate_profile_index = service.data.get(ATTR_CLIMATE_PROFILE_INDEX) - 1
+
+        for hapid in hass.data[DOMAIN]:
+            hap = hass.data[DOMAIN][hapid]
+            home = hap.home
+
+            if entity_id_list:
+                for entity_id in entity_id_list:
+                    group = hap.hmip_device_by_entity_id.get(entity_id)
+                    if group:
+                        await group.set_active_profile(climate_profile_index)
+            else:
+                for group in home.groups:
+                    if isinstance(group, AsyncHeatingGroup):
+                        await group.set_active_profile(climate_profile_index)
+
+    hass.services.async_register(
```

Comment:
Why do we need the custom service when we expose the same feature as a preset?

Reply:
First idea was to create a service to switch. 2nd idea  was given by Pascal to use profiles as preset, which is very nice. 
The service is useful, if you would'd like to switch all profiles at time.

Reply:
Can't we do that with presets too?

Reply:
Hmmh, the service is based on the profile index, that is also displayed in the app.
e.G Switch to Profile 2

The presets are name based. So we switch the preset to e.g. to "summer".
The profiles names may be diffrent  between climate entities, so this won't work good.

Reply:
Ok.